contracts/DaiBackstopSyndicateV1.sol

```diff
@@ -0,0 +1,186 @@
+
+  // The backstop price is 100 Dai for 1 MKR.
+  uint256 internal constant _MKR_BACKSTOP_BID_PRICE_DENOMINATED_IN_DAI = 100;
+
+  IERC20 internal constant _DAI = IERC20(
+    0x6B175474E89094C44Da98b954EedeAC495271d0F
+  );
+
+  IERC20 internal constant _MKR = IERC20(
+    0x9f8F72aA9304c8B593d555F12eF6589cC3A579A2
+  );
+
+  constructor() public {
+    _status = Status.ACCEPTING_DEPOSITS;
+  }
+
+  function deposit(
+    uint256 daiAmount
+  ) external returns (uint256 backstopTokensMinted) {
+    require(
+      _status == Status.ACCEPTING_DEPOSITS,
+      "Cannot deposit once auctions are activated."
+    );
+
+    require(
+      _DAI.transferFrom(msg.sender, address(this), daiAmount),
+      "Could not transfer Dai amount from caller."
+    );
+
+    backstopTokensMinted = daiAmount;
+    _mint(msg.sender, backstopTokensMinted);
+  }
+
+  function withdraw(
+    uint256 backstopTokenAmount
+  ) external returns (uint256 daiRedeemed, uint256 mkrRedeemed) {
+    // Determine the % ownership. (scaled up by 1e18)
+    uint256 shareFloat = (backstopTokenAmount.mul(1e18)).div(totalSupply());
+
+    // Burn the tokens.
+    _burn(msg.sender, backstopTokenAmount);
+
+    // TODO: make sure that _activeAuctions is accurate!
+
+    // Determine the Dai currently locked in auctions.
+    uint256 daiLockedInAuctions = _activeAuctions.mul(50000 * 1e18);
+
+    // Determine the Dai currently held by the contract.
+    uint256 daiBalance = _DAI.balanceOf(address(this));
+
+    // Combine Dai locked in auctions with the balance on the contract.
+    uint256 combinedDai = daiLockedInAuctions.add(daiBalance);
+
+    // Determine the Maker currently held by the contract.
+    uint256 makerBalance = _MKR.balanceOf(address(this));
+
+    // Determine the amount of Dai and MKR to redeem based on the share.
+    daiRedeemed = combinedDai.mul(shareFloat) / 1e18;
+    mkrRedeemed = makerBalance.mul(shareFloat) / 1e18;
+
+    // Ensure that sufficient Dai liquidity is currently available to withdraw.
+    require(
+      daiRedeemed <= daiBalance, "Insufficient Dai (in use in auctions)"
+    );
+
+    // Redeem the Dai and MKR.
+    require(_DAI.transfer(msg.sender, daiRedeemed), "Dai redemption failed.");
+    require(_MKR.transfer(msg.sender, mkrRedeemed), "MKR redemption failed.");
+  }
+
+  function activate() external {
+    require(
+      _status == Status.ACCEPTING_DEPOSITS,
+      "Cannot activate again after a prior activation."
+    );
+
+    require(
+      block.timestamp >= _AUCTION_START_TIME,
+      "Cannot activate until MKR auctions have started."
+    );
+
+    // Determine the Dai currently held by the contract.
+    uint256 daiBalance = _DAI.balanceOf(address(this));
+
+    // Jump straight to deactivated if there is not enough dai to make auctions.
+    if (daiBalance < 50000 * 1e18) {
+      _status = Status.DEACTIVATED;
+    } else {
+      // Set the status as active.
+      _status = Status.ACTIVATED;
+    }
```

Comment:
Why do we deactivate here? If we lose an auction, do we get the Dai back? In which case we would need to re-activate

Reply:
deactivation doesn't prevent withdrawals (you can always withdraw as long as the funds aren't being used in an auction), only entering new auctions

Reply:
If we hit this point, we are basically saying that the backstop failed and everything is aborded?

Reply:
I feel like changing it to ;

```solidity
if (daiBalance < 50000 * 1e18) {
  _status = Status.Status.ACCEPTING_DEPOSITS;
}
```

Would be more powerful and and secure. We could even deploy new instances of this contract while the auction is live and people could try to get to 50k then participate in auction. Doesn't allow for sandwiches either.

Reply:
Which makes me think, what if we have 40k DAI in and Bob deposits 10k DAI, activate it and then withdraw its 10k *after* it was activated. What happens?